src/video/wayland/SDL_waylandwindow.c

```diff
@@ -2537,6 +2593,17 @@ bool Wayland_CreateWindow(SDL_VideoDevice *_this, SDL_Window *window, SDL_Proper
         data->surface_status = WAYLAND_SURFACE_STATUS_SHOWN;
     }
 
+    if (data->shell_surface_type == WAYLAND_SURFACE_XDG_TOPLEVEL || data->shell_surface_type == WAYLAND_SURFACE_LIBDECOR) {
+        const char *window_id = SDL_GetStringProperty(create_props, SDL_PROP_WINDOW_CREATE_WAYLAND_WINDOW_ID_STRING, NULL);
+        if (window_id && *window_id == '\0') {
+            window_id = NULL;
+        }
+        if (window_id) {
+            data->window_id = SDL_strdup(window_id);
+            Wayland_CreateSession(c);
```

Comment:
This just defers the creation of the session object until a window with a valid ID actually requires it. This removes the requirement that the client set the session ID property before initializing the video subsystem, as clients tend to just initialize all of their required subsystems at once during early initialization, and it may need the IO subsystem initialized to retrieve the serialized session ID string from storage.

Reply:
Ack.